src/apps/sources/redux/actions.ts

```diff
@@ -14,6 +14,18 @@ export const editSourceAction = createActionThunk(
     EDIT_SOURCE_ACTION,
     api.update);
 
+export const resetCreateSourceAction = () => {
+    return (dispatch: Function) => dispatch(resetAction(CREATE_SOURCE_ACTION));
+      }
+    
+export const resetEditSourceAction = () => {
+    return (dispatch: Function) => dispatch(resetEditAction(CREATE_SOURCE_ACTION));
+      }
+      
+useEffect(() => resetCreateSource(),
+// eslint-disable-next-line react-hooks/exhaustive-deps
+[]);
```

Comment:
The `useEffect()` part belong in the `CreateSourcePage.tsx` and `EditSourcePage.tsx`, not in the `actions.ts` file. This is because the `useEffect()` call ensures that we call the action when the page is first loaded.